edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -38,31 +43,30 @@ public AvailabilityMetrics(IMetricsProvider metricsProvider, ISystemTime time)
                 "The amount of time the module was specified in the deployment",
                 new List<string> { "module_name" });
 
-            this.time = time;
-            this.availabilities = new List<Availability>();
-            this.edgeAgent = new Lazy<Availability>(() => new Availability("edgeAgent", this.CalculateEdgeAgentDowntime(), this.time));
+            this.updateCheckpointFile = new PeriodicTask(this.UpdateCheckpointFile, this.checkpointFrequency, this.checkpointFrequency, this.log, "Checkpoint Availability");
         }
 
         public void ComputeAvailability(ModuleSet desired, ModuleSet current)
```

Comment:
This should be async method and the caller in Agent.cs should await just before that method finished.